Eigen 5.0 update

ust testing newly released Eigen 5.0:

https://libeigen.gitlab.io/releases/5.0/

There were some compile failures for orca-jedi such as

/home/users/david.davies/cylc-run/mo-bundle-1019/share/mo-bundle/oops/src/oops/generic/HtlmCalculator.h:108:38: error: 'Eigen::BDCSVD<MatrixType, Options>::BDCSVD(Index, Index, unsigned int) [with MatrixType_ = Eigen::Matrix<double, -1, -1>; int Options_ = 0; Index = long int]' is deprecated: Options should be specified using the class template parameter. [-Werror=deprecated-declarations]
  108 |   linearErrorVector_(ensembleSize_), SVD_(vectorSize_, vectorSize_, Eigen::ComputeThinU),
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/users/david.davies/cylc-run/mo-bundle-1019/share/mobbs/util/spice_gnu/include/eigen3/Eigen/SVD:40,
                 from /home/users/david.davies/cylc-run/mo-bundle-1019/share/mobbs/util/spice_gnu/include/eigen3/Eigen/Dense:5,
                 from /home/users/david.davies/cylc-run/mo-bundle-1019/share/mo-bundle/oops/src/oops/mpi/mpi.h:13,
                 from /home/users/david.davies/cylc-run/mo-bundle-1019/share/mo-bundle/oops/src/oops/runs/Run.h:20,
                 from /home/users/david.davies/cylc-run/mo-bundle-1019/share/mo-bundle/orca-jedi/src/mains/orcamodel3DVar.cc:7:

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,7 +65,7 @@ endif()
 if( (CMAKE_C_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_ID MATCHES "GNU") OR ENABLE_WARNINGS_TO_ERRORS )
   message(STATUS "Forcing -Werror (treating warnings as errors)")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-deprecated-declarations")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Werror")
 endif()
 

